### ISSUE TITLE
Session : interception script should be informed about state change and when session ended by expiration

### ISSUE THREAD

**Issue by yuriyz** [GluuFederation/oxAuth#1353]
## Describe the issue
Session interception script : 
- should be informed about state change (authenticated/unauthenticated)
- when session ended by expiration, script should be invoked

**Comment by yuriyz** [comment]
It's going to be implemented with more generic method based on events instead of hard coded methods
```
def onEvent(EventType type, Event event)

with different types
enum EventType {AUTHENTICATED, UNAUTHENTICATED, EXTENDED}
... 
Goal is to make it easier to maintain. On new event we will be able to add new type : 

- no new method
- no upgrade updates

- 

```

**Comment by mtgluu** [comment]
Thanks. There should also be an enum type for timeouts correct?

**Comment by yuriyz** [comment]
There will be additional TIMEOUT type or otherwise call to existing `endSession`.

**Comment by yuriyz** [comment]
Implemented in 4.2. Docs updated.
